website/archiver/tasks.py

```diff
@@ -201,6 +201,8 @@ def archive_addon(addon_short_name, job_pk, stat_result):
     addon_name = addon_short_name
     if 'dataverse' in addon_short_name:
         addon_name = 'dataverse'
+        revision = 'latest' if addon_short_name.split('-')[-1] == 'draft' else 'latest-published'
```

Comment:
Agreed! I was thinking that if we get around to refactoring this, that we should treat draft and published as separate addon instances throughout the code base, in both WaterButler and OSF. As it is they are in an awkward in between. Which I assume is what led to this bug in the first place.